### PR TITLE
gha: automatic rules update

### PR DESCRIPTION
**Pull request by habibayassin**
@vvbandeira the following variables should be added to the gha secrets in the project's settings:
CREDS_FILE --> has the json object for the firebase creds 
API_BASE_URL --> has the Backend (with latest login flow) link with port identified without a backslash at the end ex. https://dashboard.theopenroadproject.org:8443

### CODE REVIEW

File: .github/workflows/github-actions-update-rules-force.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          CREDS_FILE: ${{ secrets.CREDS_FILE }}
+        run: python flow/util/updateRules.py --keyFile "${CREDS_FILE}" --overwrite
+      - name: Push updated rules
+        id: remote-update
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          echo "::set-output name=has_update::$(git --no-pager diff --name-only origin/master..HEAD)"
+          git add .
+          git commit -m "flow: update rules based on new set golden"
```
Comment (vvbandeira) [inline .github/workflows/github-actions-update-rules-force.yml]:
```suggestion
          git commit -m "flow: update rules based on new golden reference"
```

File: .github/workflows/github-actions-update-rules-force.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - if: "steps.remote-update.outputs.has_update != ''"
+        name: Create Draft PR
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ github.token }}
+          signoff: true
+          delete-branch: true
+          title: "Update rules"
```
Comment (vvbandeira) [inline .github/workflows/github-actions-update-rules-force.yml]:
```suggestion
          title: "[BOT] Update rules"
```

File: .github/workflows/github-actions-update-rules.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          CREDS_FILE: ${{ secrets.CREDS_FILE }}
+        run: python flow/util/updateRules.py --keyFile "${CREDS_FILE}"
+      - name: Push updated rules
+        id: remote-update
+        run: |
+          git config --local user.email "github-actions[bot]@users.noreply.github.com"
+          git config --local user.name "github-actions[bot]"
+          echo "::set-output name=has_update::$(git --no-pager diff --name-only origin/master..HEAD)"
+          git add .
+          git commit -m "flow: update rules based on new set golden"
```
Comment (vvbandeira) [inline .github/workflows/github-actions-update-rules.yml]:
```suggestion
          git commit -m "flow: update rules based on new golden reference"
```

File: .github/workflows/github-actions-update-rules.yml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      - if: "steps.remote-update.outputs.has_update != ''"
+        name: Create Draft PR
+        uses: peter-evans/create-pull-request@v5
+        with:
+          token: ${{ github.token }}
+          signoff: true
+          delete-branch: true
+          title: "Update rules"
```
Comment (vvbandeira) [inline .github/workflows/github-actions-update-rules.yml]:
```suggestion
          title: "[BOT] Update rules"
```

File: .github/workflows/github-actions-update-rules.yml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+name: Create draft PR for updated rules
+on:
+  push:
+  repository_dispatch:
+    types:
+      - set-new-golden
```
Comment (vvbandeira) [inline .github/workflows/github-actions-update-rules.yml:5]: Can't we have multiple types in one GHA and check for which type triggered the dispatch to see if we need to use `--overwrite` or not? Or some other kind of mechanism that avoids having two nearly identical GHA files?

File: flow/util/genRuleFile.py [hunk 6]
```diff
@@ -1,283 +1,330 @@
...
+    else:
+        genRuleFileFunc(designDir, # design directory
+                        False, # update
+                        True, # tighten
+                        False, # failing
+                        variant, # variant
+                        golden_metrics # metrics needed for update, default is {} in case of file
+                        )
+
+def genRuleFileFunc(design_dir, update, tighten, failing, variant, golden_metrics={}):
```
Comment (vvbandeira) [inline flow/util/genRuleFile.py]:
```suggestion
def gen_rule_file(design_dir, update, tighten, failing, variant, golden_metrics={}):
```

Comment (vvbandeira) [reply flow/util/genRuleFile.py]: For consistency

**Review by vvbandeira — CHANGES_REQUESTED** [review]

**Comment by maliberty** [comment]
@habibayassin @vvbandeira still relevant?

File: .github/workflows/github-actions-update-rules.yml [hunk 7]
```diff
@@ -0,0 +1,59 @@
+name: Create draft PR for updated rules
+on:
+  push:
```
Comment (vvbandeira) [inline .github/workflows/github-actions-update-rules.yml]: @habibayassin 
Is push required here or is it from the testing phase?

**Pull request merged**